front/tests/008-train-schedule.spec.ts

```diff
@@ -0,0 +1,110 @@
+  const homePage = new HomePage(page);
+  await homePage.goToHomePage();
+  selectedLanguage = await homePage.getOSRDLanguage();
+
+  // Navigate to the created scenario page
+  await page.goto(
+    `/operational-studies/projects/${project.id}/studies/${study.id}/scenarios/${scenario.id}`
+  );
+});
+
+test.describe('Verifying that all elements in the train schedule are loaded correctly', () => {
+  test('should accurately load the trains with their corresponding simulation results', async ({
+    page,
+  }) => {
+    test.setTimeout(120000);
+    const scenarioPage = new ScenarioPage(page);
+    const opTimetablePage = new OperationalStudiesTimetablePage(page);
+
+    // Check if the infrastructure is loaded
+    await scenarioPage.checkInfraLoaded();
+
+    // Verify the train count and various elements on the timetable page
+    await opTimetablePage.verifyTrainCount(20, selectedLanguage);
+    await opTimetablePage.verifyInvalidTrainsMessageVisibility(selectedLanguage);
+    await opTimetablePage.checkSelectedTimetableTrain();
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'Valid', 16);
+    await opTimetablePage.verifyEachTrainSimulation();
+  });
+
+  test('should accurately filter the imported trains', async ({ page }) => {
+    test.setTimeout(120000);
+    const scenarioPage = new ScenarioPage(page);
+    const opTimetablePage = new OperationalStudiesTimetablePage(page);
+
+    // Check if the infrastructure is loaded
+    await scenarioPage.checkInfraLoaded();
+
+    // Verify the train count and apply various filters
+    await opTimetablePage.verifyTrainCount(20, selectedLanguage);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'Invalid', 4);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'All', 20);
+    await opTimetablePage.filterHonoredAndVerifyTrainCount(selectedLanguage, 'Honored', 12);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'Valid', 12);
+    await opTimetablePage.filterHonoredAndVerifyTrainCount(selectedLanguage, 'Not honored', 4);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'Invalid', 0);
+    await opTimetablePage.filterHonoredAndVerifyTrainCount(selectedLanguage, 'All', 4);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'All', 20);
+
+    // Define the composition filters and verify each filter
+    const compositionFilters = [
+      { code: 'MA100', count: 7 },
+      { code: 'HLP', count: 3 },
+      { code: 'E32C', count: 1 },
+      { code: 'Without code', count: 9 },
+    ];
+
+    /* eslint-disable no-restricted-syntax, no-await-in-loop */
+    for (const filter of compositionFilters) {
+      await opTimetablePage.clickCodeCompoTrainFilterButton(
+        selectedLanguage,
+        filter.code,
+        filter.count
+      );
+    }
+    /* eslint-enable no-restricted-syntax, no-await-in-loop */
```

Comment:
I found this solution to get rid of the warnings, but it may not be the best one

```suggestion
await compositionFilters.reduce(async (promise, filter) => {
      // Wait for the previous promise to complete
      await promise;
      // Now perform the current asynchronous operation
      return opTimetablePage.clickCodeCompoTrainFilterButton(
        selectedLanguage,
        filter.code,
        filter.count
      );
    }, Promise.resolve());
```